### ISSUE TITLE
Can we take advantage of the Vector API for text analysis?

### ISSUE THREAD

**Issue by jpountz** [apache/lucene#12393]
### Description

`StandardTokenizer` is likely our most widely used tokenizer, and is reported as the main bottleneck for indexing in our nightly benchmarks, see e.g. top 5 CPU users for the 1kB Wikipedia corpus on yesterday's run:

```
PERCENT       CPU SAMPLES   STACK
10.54%        51017         org.apache.lucene.analysis.standard.StandardTokenizerImpl#getNextToken()
6.99%         33852         org.apache.lucene.index.IndexingChain$PerField#invertTokenStream()
6.47%         31309         org.apache.lucene.index.TermsHashPerField#writeByte()
5.00%         24183         org.apache.lucene.util.BytesRefHash#equals()
4.38%         21215         java.lang.Character#codePointAtImpl()
```

Intuitively, this kind of workload is amenable to vectorization, could we take advantage of vectorization to speed up text analysis and thus indexing?

**Comment by rmuir** [comment]
> Intuitively, this kind of workload is amenable to vectorization, could we take advantage of vectorization to speed up text analysis and thus indexing?

I don't think it is: each character cannot be processed in isolation, they are processed in context. https://unicode.org/reports/tr29/#Word_Boundary_Rules

**Comment by mkhludnev** [comment]
Noob says: Tokenizers for word embeddings https://github.com/huggingface/tokenizers are quite different to ours. 
`thanks to the Rust implementation. Takes less than 20 seconds to tokenize a GB of text on a server's CPU.`
And it _might be_ related to Rust's vectorization.
